Libraries/PushNotificationIOS/RCTPushNotificationManager.m

```diff
@@ -160,7 +172,6 @@ - (void)handleRemoteNotificationsRegistered:(NSNotification *)notification
   if ([app respondsToSelector:@selector(registerUserNotificationSettings:)]) {
     UIUserNotificationSettings *notificationSettings = [UIUserNotificationSettings settingsForTypes:(NSUInteger)types categories:nil];
     [app registerUserNotificationSettings:notificationSettings];
-    [app registerForRemoteNotifications];
```

Comment:
It doesn't. 
When user registers for the notification settings, the method in AppDelegate (https://github.com/facebook/react-native/pull/4277/files#diff-c4441933746f851b6a89a445751b812cR43) would trigger the didRegisterUserNotificationSettings: which calls the registerForRemotNotifications: